### DISCUSSION TITLE
[AuthZ] 视图掩码不生效

### DISCUSSION THREAD

**Discussion by fuhaiq** [apache/kyuubi#3565] [Q&A] [answered]
版本：1.6.0
AuthZ版本：Master（日期 2022-09-27）

我在 hive 里面创建了一个 view，后面关联着一些表，然后我直接在 view 的某个字符串字段上面做了一个 masking，策略为 hash。

查询应该显示为哈希码，但是未起作用，其他掩码策略均不生效。但是在 hive-server2 上是没问题的

我分别看了 kyuubi, driver 日志，没有报错

**Comment by yaooqinn** [comment]
可以试下 master， 应该支持了 cc @bowenliang123

**Reply by fuhaiq** [reply]
谢谢，我 2022-09-27 尝试的，应该是最新的了

**Reply by bowenliang123** [reply]
稍微理解一下，这是在hive创建了一个永久视图，然后在ranger上定义对这个视图某些列进行data masking的规则，在Kyuubi+Spark下未生效是吗。可不可以讲完整的细节沉淀到一个issue上进行呈现，方便大家跟踪和观察？包括所有涉及的组件和版本。

**Reply by bowenliang123** [reply]
我初步想到的是可能跟spark版本可能也有关系。3.1及之前有可能未将执行计划解析到列。

**Reply by fuhaiq** [reply]
哦，好的，我稍微整理下，稍后提到 issue 上

**Reply by bowenliang123** [reply]
好谢谢，建议可以参考下这个issue，#3515 。很具体的细节然后也有完整的上下文和现象，非常感谢。

**Reply by bowenliang123** [reply]
@fuhaiq 你先继续提issue，然后我草拟了一个应对的pr #3576 ，你提了issue之后我关联上去。我还需要一些时间准备ut进行验证。整个修复可能比预想的更复杂，在改写rowfilter之后output中属性的id对不上了。

**Reply by bowenliang123** [reply]
issue 提到这里了 #3581 ，在 pr #3576 完成了修正和单元测试的验证。

**Comment by bowenliang123** [comment, marked as answer]
@fuhaiq 你好，你反馈的这个问题已经在pr #3581 上完成解决，并合并到master分支了，请你在master重新构建authz插件再次验证下。

**Reply by fuhaiq** [reply]
太谢谢啦~
十一过节，把提 issue 的事情忘了 😭

**Reply by bowenliang123** [reply]
好，你用master构建验证下。
